frontend/readline.c

```diff
@@ -37,10 +43,39 @@ static void sigint_handler(int n) {
 }
 #endif
 
+
+static void mkdirs(const char *path) {
+	struct stat st = {0};
+	if (stat(path, &st) == 0 && S_ISDIR(st.st_mode))
+		return;
+
+	char *subpath;
+
+	subpath = strdup(path);
+	dirname(subpath);
+
+
+	if (strlen(subpath) > 1)
+		mkdirs(subpath);
+	printf("mkdir %s\n", path);
+	mkdir(path, 0700);
+	free(subpath);
+}
+
 static const char *get_history_path(void) {
 	static char history_path[PATH_MAX];
-	snprintf(history_path, sizeof(history_path),
-			"%s/.mrsh_history", getenv("HOME"));
+	if (getenv("XDG_DATA_HOME")) {
+		snprintf(history_path, sizeof(history_path),
+				"%s/mrsh", getenv("XDG_DATA_HOME"));
+	} else {
+		snprintf(history_path, sizeof(history_path),
+				"%s/.local/share/mrsh", getenv("HOME"));
+	}
+
+	mkdirs(history_path);
+
+	strncat(history_path, "/history", sizeof(history_path) - strlen(history_path) - 1);
```

Comment:
No need for a separate `strncat` here, we can just move `/history` to the `snprintf` calls above.

Reply:
I don't think so, because then I would create a directory named `history` in the path where the history file should be put.

Reply:
Oh, right, there's a call to `mkdirs` before. Well, in any case, we don't use `PATH_MAX` anymore, so when rebasing against master this logic will need to be changed…